modules/agar-sw/src/main/ts/core/ServiceWorker.ts

```diff
@@ -82,11 +85,19 @@ const sendRequestToClient = async (client: Client, request: Request): Promise<Re
           }
         });
 
-        resolve(new Response(reader, {
-          status: headData.status,
-          statusText: headData.statusText,
-          headers: headData.headers
-        }));
+        if (canHaveBody(headData.status, request.method)) {
+          resolve(new Response(reader, {
+            status: headData.status,
+            statusText: headData.statusText,
+            headers: headData.headers
+          }));
+        } else {
+          resolve(new Response(null, {
+            status: headData.status,
+            statusText: headData.statusText,
+            headers: headData.headers
```

Comment:
Since the contents are the same, why not just change whether `null` or `reader` is passed in based on `canHaveBody`?